### ISSUE TITLE
Bug: [explicit-module-boundary-types] doesn't report on class' `accessor` methods with missing explicit return and argument types

### ISSUE THREAD

**Issue by ronami** [typescript-eslint/typescript-eslint#10802]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=KYDwDg9gTgLgBAYwDYEMDOa4BU4G8BQccA9MXFMJLMACaLQUIxICehcAZhBHALxwAKAJR8AfHjgBffO1JwAdhHgUqMWnABGAV3hoAFhC1I6G4OxQIEwDNE0oofQSN7jcU-NKA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6RAD2MbMroFsA9gBMEiWgCNBsJsICG0QrSKlUGSImjRB0SODABfEAaA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
export class T {
  // reported correctly
  foo = () => { }

  // not reported but should be
  accessor bar = () => { }
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/explicit-module-boundary-types": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected the rule to report on the `accessor` property assignment, similar to a standard property.

### Actual Result

The `accessor` property didn't get reported.

### Additional Info

See https://github.com/typescript-eslint/typescript-eslint/issues/10763#issuecomment-2634900340.
